### DISCUSSION TITLE
The normalized cache in this REPRO is not updating an entity field

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#2368] [Q&A] [answered]
The normalized caching in this REPRO is not updating an entity field.

### Steps:

1. Go to https://stackblitz.com/edit/sveltejs-kit-template-default-6tzo2i
1. open a new terminal with the `+` icon
1. start apollo-server with: `cd server && npm i && node .`
1. Go to "Teams" page
1. click on the first Team ID
1. on the team page the players list is shown
1. obviously the player's name is `undefined` because in that query I don't ask for it
1. click on the first player button
1. the query is correctly providing the `TeamPlayer` with ID `0` and it's `name` but `urql` doesn't update the player `name` which is still `undefined`

### QUESTION

Why is the player `name` not updated after the manual-with-button query?

**Is this a bug?**

### GIF

![issue](https://user-images.githubusercontent.com/41120635/160656468-673ded77-38f7-4ef0-9d92-fe2e0d6ac226.gif)

![image](https://user-images.githubusercontent.com/41120635/160648009-f87efef1-9503-4969-8498-82e56cb265b6.png)

**Comment by kitten** [comment, marked as answer]
If I'm not mistaken you're feeding the `_Player` with a `player` from the list query, right? That list query doesn't include the `name` field so it won't ever receive it 🤔

Edit: So, basically `<Player>` receives the query's `player` which is from `teamByID`'s list of player, which doesn't include the `name` field but the `_Player.svelte` file's query, while it runs and asks for the `name` field isn't used and hence the `name` is never being used

**Reply by frederikhors** [reply]
Yes this is the problem. But why?

Isn't this the point in normalized cache?

**Reply by frederikhors** [reply]
I have read and re-read your answer but I don't understand if it's my problem or urql problem or graphql problem.

**Reply by JoviDeCroock** [reply]
The overarching query, injects the player prop, isn't asking for name hence not notified by the update of the name

**Reply by frederikhors** [reply]
In my real project I need to query many fields for players which I don't want to query on team's query one.

How would you fix this?

**Reply by JoviDeCroock** [reply]
I mean... :joy: you can look at fragments etc but not really up to us 😅

**Reply by frederikhors** [reply]
@JoviDeCroock in my real project I'm using fragments and it's the same. I don't understand if it's a logic problem. Why the cache doesn't recognize the new `name` field of the Player? I don't care about name in the previous query.

**Reply by frederikhors** [reply]
The `Player.svelte` query is used with `JSON.stringify()`. And you can see the `name` field is present. What I don't understand is why the global normalized cache does not update the other query list players: I mean is the same entity! Should I use resolvers?

**Reply by frederikhors** [reply]
I think I'm understanding. Are you perhaps saying that if the original query doesn't include fields that change after (with other queries) those fields are not updated anymore? I need other queries for this behavior? Maybe on this case another query named `playersByTeamID()`?
